test/periodic.jl

```diff
@@ -0,0 +1,369 @@
+@testset "periodic" begin
+    @testset "_periodic" begin
+        @testset "$f" for f in (sin, cos)
+            # A typical 24-hour day
+            for h in 0:24
+                result = _periodic(
+                    f,
+                    ZonedDateTime(2015, 6, 1, tz"America/Winnipeg") + Hour(h),
+                    Day(1),
+                )
+                expected = f(2π / 24 * h)
+                @test result ≈ expected atol=1e-14
+            end
+
+            # A spring daylight-saving time change where the day has 23-hours
+            for h in 0:23
+                result = _periodic(
+                    f,
+                    ZonedDateTime(2015, 3, 8, tz"America/Winnipeg") + Hour(h),
+                    Day(1),
+                )
+                expected = f(2π / 23 * h)
+                @test result ≈ expected atol=1e-14
+            end
+
+            # A fall daylight-saving time change where the day has 25-hours
+            for h in 0:25
+                result = _periodic(
+                    f,
+                    ZonedDateTime(2015, 11, 1, tz"America/Winnipeg") + Hour(h),
+                    Day(1),
+                )
+                expected = f(2π / 25 * h)
+                @test result ≈ expected atol=1e-14
+            end
+        end
+
+        @testset "phase shift" begin
+            @test _periodic(sin, DateTime(1), Week(1)) == 0
```

Comment:
as we've seen below `DateTime(1)` can be weird (start of week is a Saturday)

So might be better to provide modern `DateTime`s in the range we expect to use